src/libsyntax_pos/hygiene.rs

```diff
@@ -129,13 +138,98 @@ impl HygieneData {
         }
         HYGIENE_DATA.with(|data| f(&mut *data.borrow_mut()))
     }
+
+    pub fn safe_with<T, F: FnOnce(&HygieneData) -> T>(f: F) -> T {
+        // FIXME(twk): not sure how this would behave...
```

Comment:
I couldn't find any info on how multiple declarations of `thread_local!`s behave, so this might need to be moved for the expected behaviour.